src/puppet/puppet-abstract.ts

```diff
@@ -16,6 +16,8 @@
  *   limitations under the License.
  *
  */
+import { pipe } from 'fp-ts/function'
```

Comment:
The `fp-ts` has a compatible problem with ESM, which leads this line to fail in our CI.

replace it to:

```ts
import { function as FP } from 'fp-ts'
```

and use `FP.pipe` later, can make the ESM/CI happy.

Reply:
Done